### PR TITLE
Configured alerts for Seldon and Trino 

### PR DESCRIPTION
**Pull request by ravitejamanepalli**
Added Container memory, CPU and UP alerts and configured service monitor for seldon. 
[https://github.com/operate-first/apps/issues/1677](url). 
CPU alert [https://github.com/google/cadvisor/issues/2026](url).

### CODE REVIEW

**Comment by ravitejamanepalli** [comment]
Related-to: [https://github.com/operate-first/apps/issues/1677](url)

File: kfdefs/overlays/moc/smaug/seldon/alerts.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+metadata: 
+  name: seldon-alerts
+spec: 
+  groups: 
+    - 
+      name: alert_rules
+      rules: 
+        - alert: InstanceDown
+          expr: up {namespace="opf-seldon"}== 0
+          for: 1m
```
Comment (harshad16) [inline kfdefs/overlays/moc/smaug/seldon/alerts.yaml:12]: maybe increase this to 10m or 15m
as some time, deployment is just syncing up or restarting 
so lets open alerts if for a period if its down

File: kfdefs/overlays/moc/smaug/trino/alerts.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+metadata: 
+  name: trino-alerts
+spec: 
+  groups: 
+    - 
+      name: alert_rules
+      rules: 
+        - alert: InstanceDown
+          expr: up {namespace="opf-trino", container!=""}== 0
+          for: 1m
```
Comment (harshad16) [inline kfdefs/overlays/moc/smaug/trino/alerts.yaml:12]: maybe increase this to 10m or 15m
as some time, deployment is just syncing up or restarting 
so lets open alerts if for a period if its down

**Review by harshad16 — CHANGES_REQUESTED** [review]
lgtm
just one comment

File: .aicoe-ci.yaml [hunk 3]
```diff
@@ -1,2 +1,12 @@
...
+
+build:
+  build-stratergy: Source
+  build-source-script: "image:///opt/app-root/builder"
+  base-image: quay.io/operate-first/s2i-sre-notebook:latest
+  custom-tag: latest
+  registry: quay.io
+  registry-org: operate-first
+  registry-project: hitchhikers-guide
+  registry-secret: operate-first-pusher-secret
```
Comment (harshad16) [inline .aicoe-ci.yaml:12]: this is not a part of this repo, remove it pls

File: kfdefs/overlays/moc/smaug/trino/alerts.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      name: alert_rules
+      rules:
+        - alert: InstanceDown
+          expr: up {namespace="opf-trino", container!=""}== 0
+          for: 15m
+          labels:
+            severity: critical
+          annotations:
+            summary: "Instance [{{ $labels.instance }}] down"
+            description: "[{{ $labels.instance }}] of job [{{ $labels.job }}] has been down for more than 1 minute."
```
Comment (harshad16) [inline kfdefs/overlays/moc/smaug/trino/alerts.yaml:17]: please fix the description

**Review by harshad16 — CHANGES_REQUESTED** [review]
some of new changes shouldnt be part of this pr.

**Review by mubariskhan96 — CHANGES_REQUESTED** [review]

File: kfdefs/overlays/moc/smaug/seldon/seldon-podmonitor.yaml [hunk 5]
```diff
@@ -3,9 +3,7 @@ kind: PodMonitor
 metadata:
   name: seldon-podmonitor
 spec:
-  selector:
-    matchLabels:
-      app.kubernetes.io/managed-by: seldon-core
+  selector: {}
   podMetricsEndpoints:
     - port: metrics
       path: /prometheus
```
Comment (harshad16) [inline kfdefs/overlays/moc/smaug/seldon/seldon-podmonitor.yaml:9]: Why is the namespace selector below pointing to any ?

**Review by harshad16 — CHANGES_REQUESTED** [review]
lgtm
just one question

File: kfdefs/overlays/moc/smaug/seldon/seldon-podmonitor.yaml (hunk 5 above)
Comment (ravitejamanepalli) [reply kfdefs/overlays/moc/smaug/seldon/seldon-podmonitor.yaml:9]: I think it will match entire namespace by default, there can be other services with different label where we want to get metrics in the same namespace, leaving empty will help to discover metrics in entire namespace otherwise we get restricted to specific service.

Comment (mubariskhan96) [reply kfdefs/overlays/moc/smaug/seldon/seldon-podmonitor.yaml:9]: If the namespace matches are not correct, targets won't show up. And it is also consistent with the namespace selectors defined for the servicemonitors for other services (ex: jupyterhub[https://github.com/operate-first/apps/blob/master/kfdefs/overlays/moc/smaug/jupyterhub/servicemonitor.yaml](url))

Comment (harshad16) [reply kfdefs/overlays/moc/smaug/seldon/seldon-podmonitor.yaml:9]: my concern is that the namespaceselector is set to any , so it loooking at whole cluster
if we remove the selector per label, then it would pick up all the metrics in the whole namespace.
which could causes if in future, while debugging metrics. 

i would suggest: either restrict namespace to seldon or selector based on seldon label

File: .aicoe-ci.yaml [hunk 6]
```diff
@@ -1,2 +0,0 @@
----
-check: []
```
Comment (harshad16) [inline .aicoe-ci.yaml:2]: please revert this change here.

**Review by harshad16 — CHANGES_REQUESTED** [review]
some minimal changes needed

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

File: kfdefs/overlays/moc/smaug/seldon/seldon-podmonitor.yaml (hunk 5 above)
Comment (ravitejamanepalli) [reply kfdefs/overlays/moc/smaug/seldon/seldon-podmonitor.yaml]: Leaving namespace selector to empty will help to get all metrics in the same namespace, in this way we can monitor entire namespace
